mdata/aggmetrics.go

```diff
@@ -151,6 +151,10 @@ func (ms *AggMetrics) GetOrCreate(key schema.MKey, schemaId, aggId uint16, inter
 
 	agg := Aggregations.Get(aggId)
 	confSchema := Schemas.Get(schemaId)
+	var futureTolerance uint32
+	if futureToleranceRatio > 0 {
+		futureTolerance = uint32(confSchema.Retentions.MaxRetention()) * uint32(futureToleranceRatio) / 100
```

Comment:
This is wrong.  The tolerance needs to be based on the TTL of the raw points being ingested.
The TTL is extracted from the confSchema.Retentions in NewAggMetric().
```
ret := retentions.Rets[0]
...
		ttl:             uint32(ret.MaxRetention()),
```

given that the TTL is being extracted in `NewAggMetric()` we should just pass futureToleranceRatio to NewAggMetric and set the futureTolerance there.

Reply:
Ok, I'll change that. This will then mean that the rollup AggMetric instances will have a higher future tolerance than the raw one. This is fine, as the raw one will just reject every data point beyond its own future tolerance, so those would never get passed to the rollup archives then.